containers/Makefile

```diff
@@ -59,42 +64,30 @@ $(info IMAGE_PREFIX=$(IMAGE_PREFIX))
 
 all: nginx-proxy nginx-storage concent-api postgresql verifier
 
-nginx-proxy: build/ $(NGINX_PROXY_SOURCE) build/nginx-proxy/concent-api-assets.tar build/nginx-proxy/concent-api-docs.tar
-	docker build --tag $(IMAGE_PREFIX)nginx-proxy:$(IMAGE_VERSION) build/nginx-proxy/
-	docker tag $(IMAGE_PREFIX)nginx-proxy:$(IMAGE_VERSION) $(IMAGE_PREFIX)nginx-proxy:latest
-
-nginx-storage: build/ $(NGINX_STORAGE_SOURCE)
-	docker build --tag $(IMAGE_PREFIX)nginx-storage:$(IMAGE_VERSION) build/nginx-storage/
-	docker tag $(IMAGE_PREFIX)nginx-storage:$(IMAGE_VERSION) $(IMAGE_PREFIX)nginx-storage:latest
-
-postgresql: build/ $(POSTGRESQL_SOURCE)
-	docker build --tag $(IMAGE_PREFIX)postgresql:$(IMAGE_VERSION) build/postgresql/
-	docker tag $(IMAGE_PREFIX)postgresql:$(IMAGE_VERSION) $(IMAGE_PREFIX)postgresql:latest
-
-concent-api: build/ $(CONCENT_API_SOURCE) build/concent-api/requirements.lock build/concent-api/testing.pref build/repositories/concent/
-	docker build --tag $(IMAGE_PREFIX)concent-api:$(IMAGE_VERSION) build/concent-api/
-	docker tag $(IMAGE_PREFIX)concent-api:$(IMAGE_VERSION) $(IMAGE_PREFIX)concent-api:latest
-
-verifier: build/ $(VERIFIER_SOURCE) build/repositories/golem/ build/verifier/golem/scripts/ build/verifier/golem/imgverifier-requirements.txt
-	docker build --tag $(IMAGE_PREFIX)verifier:$(IMAGE_VERSION) build/verifier/
-	docker tag $(IMAGE_PREFIX)verifier:$(IMAGE_VERSION) $(IMAGE_PREFIX)verifier:latest
+nginx-proxy: $(NGINX_PROXY_SOURCE) build/nginx-proxy/concent-api-assets.tar build/nginx-proxy/concent-api-docs.tar | build/
+	$(DOCKER_BUILD)
+	$(DOCKER_TAG)
 
-docker-clean: nginx-proxy-clean concent-api-clean postgresql-clean verifier-clean
+nginx-storage: $(NGINX_STORAGE_SOURCE) | build/
+	$(DOCKER_BUILD)
+	$(DOCKER_TAG)
 
-nginx-proxy-clean:
-	docker rmi $(IMAGE_PREFIX)nginx-proxy:$(IMAGE_VERSION) --force
+postgresql: $(POSTGRESQL_SOURCE) | build/
+	$(DOCKER_BUILD)
+	$(DOCKER_TAG)
 
-nginx-storage-clean:
-	docker rmi $(IMAGE_PREFIX)nginx-storage:$(IMAGE_VERSION) --force
+concent-api: $(CONCENT_API_SOURCE) build/concent-api/requirements.lock build/concent-api/testing.pref build/repositories/concent/ | build/
+	$(DOCKER_BUILD)
+	$(DOCKER_TAG)
 
-postgresql-clean:
-	docker rmi $(IMAGE_PREFIX)postgresql:$(IMAGE_VERSION) --force
+verifier: $(VERIFIER_SOURCE) build/repositories/golem/ build/verifier/golem/scripts/ build/verifier/golem/imgverifier-requirements.txt | build/
+	$(DOCKER_BUILD)
+	$(DOCKER_TAG)
 
-concent-api-clean:
-	docker rmi $(IMAGE_PREFIX)concent-api:$(IMAGE_VERSION) --force
+docker-clean: nginx-proxy-rmi nginx-storage-rmi postgresql-rmi concent-api-rmi  verifier-rmi
 
-verifier-clean:
-	docker rmi $(IMAGE_PREFIX)verifier:$(IMAGE_VERSION) --force
+%-rmi:
+	docker rmi $(IMAGE_PREFIX)$*:$(IMAGE_VERSION) --force
```

Comment:
Any target that does not really create a file with the corresponding name must be a [Phony target](https://www.gnu.org/software/make/manual/html_node/Phony-Targets.html).